src/AppBundle/Controller/BlogController.php

```diff
@@ -36,12 +36,18 @@ class BlogController extends Controller
     /**
      * @Route("/", name="blog_index")
      */
-    public function indexAction()
+    public function indexAction(Request $request)
     {
-        $em = $this->getDoctrine()->getManager();
-        $posts = $em->getRepository('AppBundle:Post')->findLatest();
```

Comment:
I'm sorry but I don't understand your comment. Maybe KnpPaginator behaves differently, but I usually put all this paginator logic inside the specific method of the custom repository. In the controller I only get the `page` number and pass it to the query.